src/components/DimensionMenu.js

```diff
@@ -42,97 +24,158 @@ const getRemoveMenuItem = onClick => (
 
 const getDividerItem = key => <Divider light key={key} />
 
-export const DimensionMenu = ({
-    dimensionId,
+const getDualAxisMenuItemLabel = (
     currentAxisId,
     visType,
-    numberOfDimensionItems,
-    dualAxisItemHandler,
-    axisItemHandler,
-    removeItemHandler,
-    anchorEl,
-    onClose,
-}) => {
-    const menuItems = []
-
-    const isDimensionInLayout = !!currentAxisId
-
-    // dual axis item
-    const hasDualAxis = shouldHaveDualAxisOption({
-        dimensionId,
-        currentAxisId,
-        visType,
-        numberOfDimensionItems,
-    })
-
-    // add/move to axis item
-    const availableAxisIds = getAvailableAxes(visType)
-
-    const applicableAxisIds = availableAxisIds.filter(
-        axisId => axisId !== currentAxisId
-    )
-
-    // create menu items
-    if (hasDualAxis) {
-        menuItems.push(
-            getDualAxisItem(dimensionId, () => {
-                dualAxisItemHandler()
-                onClose()
-            })
-        )
+    numberOfDimensionItems
+) => {
+    let label
+
+    if (!isDualAxisType(visType)) {
```

Comment:
This could be solved in a more compact / ES6 way and possibly be merged with the logic check on line 100. @janhenrikoverland thoughts?